minidump-processor/src/op_analysis.rs

```diff
@@ -153,62 +154,174 @@ mod amd64 {
+        ) -> Result<(), OpAnalysisError> {
+            // Shortcut -- If the instruction doesn't access memory, just return an empty list
+            let mem_size = match decoded_instruction.mem_size() {
+                Some(access) => access.bytes_size(),
+                None => return Ok(()),
             };
 
+            for idx in 0..decoded_instruction.operand_count() {
+                let operand = decoded_instruction.operand(idx);
+
+                let maybe_access = match operand {
+                    Operand::DisplacementU32(disp) => Some(MemoryAccess {
+                        address: disp.into(),
+                        is_likely_null_pointer_dereference: false,
+                        size: mem_size,
+                    }),
+                    Operand::DisplacementU64(disp) => Some(MemoryAccess {
+                        address: disp,
+                        is_likely_null_pointer_dereference: false,
+                        size: mem_size,
+                    }),
+                    other_operand => {
+                        if let Some(op_info) = RegOperandInfo::try_from_operand(other_operand) {
+                            Some(self.calculate_address(mem_size, op_info)?)
+                        } else {
+                            None
+                        }
+                    }
+                };
+
+                if let Some(access) = maybe_access {
+                    accesses.push(access);
+                }
+            }
+
+            Ok(())
+        }
+
+        /// Indirect memory accesses include instructions which change the instruction pointer.
+        fn indirect(
+            &self,
+            accesses: &mut Vec<MemoryAccess>,
+            decoded_instruction: &Instruction,
+        ) -> Result<(), OpAnalysisError> {
+            match decoded_instruction.opcode() {
+                Opcode::CALL | Opcode::CALLF | Opcode::JMP | Opcode::JMPF | Opcode::JMPE => {
+                    if decoded_instruction.operand_count() != 1 {
+                        tracing::warn!("call/jmp instruction had incorrect operand count");
+                        return Ok(());
+                    }
+                    // We assume that relative offsets (for CALL, JMP) and absolute values (CALLF,
+                    // JMPF) will be valid, so we don't check immediate operands, only registers.
+                    if let Operand::Register(reg) = decoded_instruction.operand(0) {
+                        let address = self.get_reg(reg)?;
+                        accesses.push(MemoryAccess {
+                            address,
+                            is_likely_null_pointer_dereference: address == 0,
+                            size: Some(1),
+                        });
+                    }
+                }
+                // TODO: if the full minidump information is plumbed down here, we can support
+                // RET/IRET as well (by reading the stack to get the return address)
```

Comment:
It would be nice to use the stack information for this. The stack info could also be used to read any indirect calls (when they fall in stack memory), like `call [rax+8]`, which is probably common for vtables. If the minidump has heap memory that could be used too!

Reply:
This is a worthy follow-up, I'll file an issue for it. I was considering doing it for `ret` but you're right that it might be more interesting to deal with vtables, especially those of dead objects which would automatically lead to a non-canonical access due to the poison pattern.